### ISSUE TITLE
http() destination: time_reopen() option not activated

### ISSUE THREAD

**Issue by mitzkia** [balabit/syslog-ng#1500]
syslog-ng version:
 * builded from actual OSE master 
 * latest git commit: 282ec7c521918d79a6a01189f0c7d8d0bc9ac26c

Reproduction:
1, Use the following (valid) syslog-ng config (http server did not listening):
```
@version: 3.9

options {
    time_reopen(1);
};

source s_network {
    network( ip("127.0.0.1") port(10001));
};

destination d_http {
    http(
        method(put)
        url("http://127.0.0.1:8080")
    );
};


log {
    source(s_network);
    destination(d_http);
};
```
2, syslog-ng should send an error message to console log when time_reopen() is activated, but it is missing:
```
$ sbin/syslog-ng -Fe -f ~/micek_work/syslog_ng_configs/syslog_ng_http_3.conf 
syslog-ng: Error setting capabilities, capability management disabled; error='Operation not permitted'
[2017-05-29T20:14:52.651916] WARNING: Configuration file format is newer than the current version, please specify the current version number (3.9) in the @version directive. syslog-ng will operate at its highest supported version in this mode;
[2017-05-29T20:14:52.676323] syslog-ng starting up; version='3.9.1'
^C[2017-05-29T20:14:55.898044] syslog-ng shutting down; version='3.9.1'

```
* similar message from network destination
```
[2017-05-29T20:11:28.882633] Syslog connection failed; fd='13', server='AF_INET(127.0.0.1:9999)', error='Connection refused (111)', time_reopen='10'
```

**Comment by mitzkia** [comment]
Some further notes. 
If we start syslog-ng with an active http destination, and send 1 message to syslog-ng it will transfer it to http destination correctly. But if the http destination goes down, and syslog-ng gets new messages the  time_reopen() will not activated, but the retries() yes. So messages will be dropped automatically.

```
[2017-05-31T09:53:12.247531] Syslog connection accepted; fd='15', client='AF_INET(127.0.0.1:38376)', local='AF_INET(127.0.0.1:10001)'
[2017-05-31T09:53:12.247763] Incoming log entry; line='<38>2017-05-31T09:53:12 localhost prg00000[1234]: seq: 0000000000, thread: 0000, runid: 1496217192, stamp: 2017-05-31T09:53:12 PADDPADDPADDPADDPADDPADDPADDPADDPADDPADDPADDPADDPADDPADDPADDPADDPADDPADDPADDPADDPADDPADDPADDPADDPADDPADDPADDPADDPADDPADDPADDPADD'
[2017-05-31T09:53:12.247878] Setting value; msg='0x7f0dac014990', name='HOST_FROM', value='localhost'
[2017-05-31T09:53:12.247914] Setting value; msg='0x7f0dac014990', name='HOST', value='localhost'
[2017-05-31T09:53:12.247959] Setting value; msg='0x7f0dac014990', name='SOURCE', value='s_network'
[2017-05-31T09:53:12.248023] EOF occurred while reading; fd='15'
[2017-05-31T09:53:12.248123] Syslog connection closed; fd='15', client='AF_INET(127.0.0.1:38376)', local='AF_INET(127.0.0.1:10001)'
[2017-05-31T09:53:12.248271] Closing log transport fd; fd='15'
[2017-05-31T09:53:12.248511] curl: error sending HTTP request; error='Couldn\'t connect to server'
[2017-05-31T09:53:15.252008] curl: error sending HTTP request; error='Couldn\'t connect to server'
[2017-05-31T09:53:18.255454] curl: error sending HTTP request; error='Couldn\'t connect to server'
[2017-05-31T09:53:18.255496] Multiple failures while sending message to destination, message dropped; driver='d_http2#0', number_of_retries='3'
^C[2017-05-31T09:53:27.691006] syslog-ng shutting down; version='3.9.1'
[2017-05-31T09:53:27.791334] Worker thread finished; driver='d_http2#0'
[2017-05-31T09:53:27.791782] Running application hooks; hook='4'
```

**Comment by mitzkia** [comment]
Ok, the #1540 fixes this issue.
